src/js/todo.js

```diff
@@ -0,0 +1,85 @@
+            <div class="todo">
+              <input type="checkbox" class="check-box" value="${activitie.completed}" ${activitieCheck}>
+              <input type="text" class="list-description" value="${activitie.desc}">
+              <i class="fa-solid fa-ellipsis-vertical"></i>
+              <button type="button" class="delete-btn"><i class="fa-solid fa-trash"></i></button>
+            </div>`;
+    });
+  }
+
+  for (let i = 0; i < todoDiv.querySelectorAll('.todo').length; i += 1) {
+    todoDiv.querySelectorAll('.todo')[i].querySelector('.check-box').addEventListener('click', () => {
+      activities[i].completed = !activities[i].completed;
+      localStorage.setItem('todo', JSON.stringify(activities));
+    });
+
+    todoDiv.querySelectorAll('.todo')[i].addEventListener('focusin', () => {
+      todoDiv.querySelectorAll('.todo')[i].classList.add('active');
+      todoDiv.querySelectorAll('.todo')[i].querySelector('.list-description').classList.add('active');
+      todoDiv.querySelectorAll('.todo')[i].querySelector('.delete-btn').style.display = 'block';
+      todoDiv.querySelectorAll('.todo')[i].querySelector('.fa-ellipsis-vertical').style.display = 'none';
+    });
+
+    todoDiv.querySelectorAll('.todo')[i].querySelector('.delete-btn').addEventListener('click', () => {
+      activities.splice(i, 1);
+      for (let a = i; a < activities.length; a += 1) {
+        activities[a].index -= 1;
+      }
+      index -= 1;
+      localStorage.setItem('todo', JSON.stringify(activities));
+      render();
+    });
+
+    todoDiv.querySelectorAll('.todo')[i].addEventListener('focusout', (e) => {
+      const parent = todoDiv.querySelectorAll('.todo')[i];
+      const leavingParent = !parent.contains(e.relatedTarget);
+      if (leavingParent) {
+        todoDiv.querySelectorAll('.todo')[i].classList.remove('active');
+        todoDiv.querySelectorAll('.todo')[i].querySelector('.list-description').classList.remove('active');
+        todoDiv.querySelectorAll('.todo')[i].querySelector('.fa-ellipsis-vertical').style.display = 'flex';
+        todoDiv.querySelectorAll('.todo')[i].querySelector('.delete-btn').style.display = 'none';
+      }
+    });
+
+    todoDiv.querySelectorAll('.todo')[i].querySelector('.list-description').addEventListener('change', (e) => {
+      activities[i].desc = e.target.value;
+      localStorage.setItem('todo', JSON.stringify(activities));
+    });
+  }
+};
+
+addText.addEventListener('keypress', (e) => {
+  if (e.key === 'Enter') {
+    if (addText.value === '') {
+      // eslint-disable-next-line no-alert
+      alert('It cant be Empty');
+    } else {
+      const todoVal = addText.value;
+      addText.value = '';
+      index += 1;
+      activities.push(
+        {
+          index: index,
+          desc: todoVal,
+          completed: false,
+        },
+      );
+      localStorage.setItem('todo', JSON.stringify(activities));
+      render();
+    }
+  }
+});
+
+window.onload = render();
```

Comment:
- [ ] For some reason, the added list of todo get disappeared on page reload. The added lists remain on the local storage thou. I think you should check how you handle your browser memory when the page loads. Look at the screenshots of what I have while testing your app.

### After adding todo

![adding](https://user-images.githubusercontent.com/26597432/199535949-1df7b9f4-6ce8-40d6-9cbc-ac208a14f617.png)

### After page reload

![after-reload](https://user-images.githubusercontent.com/26597432/199536133-af746d06-6d27-43ec-bb0d-3e3598d5c1bb.png)

Reply:
Hi @Qoosim Thank you for your feedback
There is a small typo thanks for letting me know I will take care of it

Thanks again